src/App.js

```diff
@@ -1,17 +1,33 @@
-import "./components/HeaderComponent/Header"
+import React, { Component } from 'react';
 import './App.css';
-import Header from "./components/HeaderComponent/Header";
+import HeaderComponent from "./components/HeaderComponent/Header";
 import Description from "./components/MainComponent/Description/Description";
 import Footer from "./components/FooterComponent/Footer";
 
-function App() {
-  return (
-    <div className="App">
-        <Header/>
-        <Description/>
-        <Footer/>
-    </div>
-  );
+class App extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            inputValue: "",
+            inputId: ""
+        }
+    }
+
+  handleDataChange = (inputValue, inputId) => {
+    this.setState({ inputValue, inputId});
+  };
+
+
+  render() {
+    return (
+      <div className="App">
+          <HeaderComponent mealsCount={this.state}/>
```

Comment:
It's better to pass here the number of cart items, not to have complex logic in HeaderComponent